app/models/ability.rb

```diff
@@ -0,0 +1,23 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    can :read, Post
+    can :read, Comment
+    can :create, Comment
+    can(:update, Comment, user:)
+
+    if user.is?('default') # additional permissions for logged in users (they can read their own posts)
+      can :create, Post, author: user
+      can :destroy, Post, author: user
+      can :update, Post, author: user
+
+      can :destroy, Comment, user:
+    else
+      can :create, Post
+      can :destroy, Post
+      can :update, Post
+
+      can :destroy, Comment
+    end
+  end
```

Comment:
- [ ] [Optional] Your app is working as expected and the conditions you added here are functioning well and fulfill the requirements, but it would be much better professionally if you make it like the suggestion below:
```suggestion
def initialize(user)

         user ||= User.new 

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user 
         can :manage, Comment, author: user
         can :read, :all
         end
     end
```